autofs-5.1.5 - nss workaround

From: Ian Kent <raven@themaw.net>

Pre-open nss libraries to prevent possible library unloads
that could leave library static data undefined.
---
 daemon/automount.c |    8 ++++++++
 1 file changed, 8 insertions(+)

--- autofs-5.0.5.orig/daemon/automount.c
+++ autofs-5.0.5/daemon/automount.c
@@ -2444,6 +2444,9 @@ int main(int argc, char *argv[])
 		dh_tirpc = dlopen("libtirpc.so.1", RTLD_NOW);
 #endif
 
+	void *dh_nss3 = dlopen("libnss3.so", RTLD_NOW);
+	void *dh_nspr4 = dlopen("libnspr4.so", RTLD_NOW);
+
 	master_read = master_read_master(master_list, age, 0);
 	if (!master_read) {
 		/*
@@ -2487,6 +2490,11 @@ int main(int argc, char *argv[])
 	release_flag_file();
 	macro_free_global_table();
 
+	if (dh_nss3)
+		dlclose(dh_nss3);
+	if (dh_nspr4)
+		dlclose(dh_nspr4);
+
 #ifdef TIRPC_WORKAROUND
 	if (dh_tirpc)
 		dlclose(dh_tirpc);
